android/beagle/src/main/java/br/com/zup/beagle/android/components/ListViewTwo.kt

```
@@ -172,24 +166,13 @@ internal class ListViewContextAdapter2(
 
     private fun isOrientationVertical() = (orientation == RecyclerView.VERTICAL)
 
-    private fun BeagleFlexView.configViewPosition(position: Int) {
-        val templateClone = template
-        this.addServerDrivenComponent(templateClone, this@ListViewContextAdapter2.rootView)
-        this.setContextData(ContextData("item", listItems[position]))
-        this@ListViewContextAdapter2.rootView.generateViewModelInstance<ScreenContextViewModel>().linkBindingToContextAndEvaluateThem()
-    }
-
     override fun onBindViewHolder(holder: ContextViewHolderTwo, position: Int) {
-        val item = listItems[position]
-        val view = holder.itemView
-        view.setContextData(ContextData(id = "item", value = item))
-        view.getContextBinding()?.let { contextBinding ->
-            rootView.generateViewModelInstance<ScreenContextViewModel>().notifyBindingChanges(contextBinding)
-        }
+        holder.itemView.setContextData(ContextData(id = "item", value = listItems[position]))
     }
 
     fun setList(list: List<Any>) {
         try {
+            clearList()
             listItems = ArrayList(list)
             notifyDataSetChanged()
         } catch (e: Exception) {
```

<table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/4495e63a2d6da359914aced77a623bdc61db730b/android/beagle/src/main/java/br/com/zup/beagle/android/components/ListViewTwo.kt#L178"></span>Empty catch block detected. If the exception can be safely ignored, name the exception according to one of the exemptions as per the configuration of this rule.</td>
    </tr>
  </tbody>
</table>